### PR TITLE
Make grpcio-tools compile targeting Mac OS X 10.6

### PR DESCRIPTION
**Pull request by soltanmm**
Also C++98-ify's the Python plug-in code and removes the C++11 requirement from grpcio-tools builds. Such changes aren't strictly necessary but improve compatibility in any case.

This combined with #6791 and #7196 will enable us to remove the Mac OS X 10.6-specific hackery in Python setup code and improve our wheel compatibility (which is currently... not good).

It's for discussion whether or not we consider the potential wheel compatibility issues to be GA blockers or not. The effort necessary to fix them appear minimal, and the gains arguably pretty high.

### CODE REVIEW

**Comment by soltanmm** [comment]
DNM because #6791 #6791 #6791.

File: tools/distrib/python/grpcio_tools/setup.py [hunk 1]
```diff
@@ -95,7 +95,10 @@ def protoc_ext_module():
           protoc_lib_deps.CC_INCLUDE,
       ],
       language='c++',
-      define_macros=[('HAVE_PTHREAD', 1)],
+      define_macros=[
+          ('HAVE_PTHREAD', 1,),
+          ('GOOGLE_PROTOBUF_NO_THREADLOCAL', 1,),
```
Comment (kpayson64) [inline tools/distrib/python/grpcio_tools/setup.py]: Any chance we can only set this flag if MACOS_DEPLOYMENT_TARGET < 10.7?

Comment (soltanmm) [inline tools/distrib/python/grpcio_tools/setup.py]: We could. But this is just the code generator (performance isn't super important), and the protobuf runtime already falls back to pthread-based TLS if it doesn't have access to `__thread` anyway, so I don't see any specific harm in setting it. Am I missing anything?

Comment (kpayson64) [inline tools/distrib/python/grpcio_tools/setup.py]: Your right, I was thinking about the protobuf runtime but I'm fine with it if its for code generation.

**Comment by kpayson64** [comment]
LGTM

**Comment by nathanielmanistaatgoogle** [comment]
Luhguhtum.

Many of the changes in this look formatting-related - did they come from a `clang-format` pass? Although it's not GA-targeted I remain excited about [this change](https://github.com/grpc/grpc/pull/6539).

**Comment by soltanmm** [comment]
The formatting changes were so I could read the code (tabs vs spaces and our style guide chose spaces but lo we let tabs-as-8-character-giants in) and are closer in style to what clang-format'd do anyway afaik. -.-'

**Comment by jtattermusch** [comment]
This PR is way out of date. Closing.

**Pull request closed without merging**